### PR TITLE
fix(quantic): Allow to display only the range filter when numberOfValues =0 in numeric-facet

### PR DESCRIPTION
**Pull request by aelfaleh**
This PR fix two bugs:
https://coveord.atlassian.net/browse/SFINT-4255
https://coveord.atlassian.net/browse/SFINT-4261

**numeric-facet:**
`<c-quantic-numeric-facet field="ytlikecount" label="Youtube Likes" with-input="integer" engine-id={engineId} number-of-values="0"></c-quantic-numeric-facet>`

image

**filter range numeric-facet without results:**
image

### CODE REVIEW

**Comment by nathanlb** [comment]
I'm a bit confused by the description. The bug fix should allow users to configure a numeric facet that only displays the input form and no clickable filters by setting `number-of-values="0"`.
Is that in fact what this change does?

**Comment by aelfaleh** [comment]
> I'm a bit confused by the description. The bug fix should allow users to configure a numeric facet that only displays the input form and no clickable filters by setting `number-of-values="0"`. Is that in fact what this change does?

I tried to create numeric facet with `number-of-values` equal to `0`, but i couldn't. I think I misunderstand the bug. let me try again 😆

File: packages/quantic/cypress/integration/facets/numeric-facet/numeric-facet.cypress.ts [hunk 1]
```diff
@@ -190,6 +193,22 @@ describe('Numeric Facet Test Suite', () => {
...
+        const max = 10000000000;
+
+        visitNumericFacetPage(customWithInputSettings);
+
+        Actions.inputMinValue(min);
+        Actions.inputMaxValue(max);
+        Actions.submitManualRange();
+
+        Expect.displayValues(false);
+        // Expect.search.numberOfResults(0);
```
Comment (nathanlb) [inline packages/quantic/cypress/integration/facets/numeric-facet/numeric-facet.cypress.ts:208]: What's up with this?

File: packages/quantic/force-app/main/default/lwc/quanticNumericFacet/quanticNumericFacet.js [hunk 2]
```diff
@@ -239,6 +248,7 @@ export default class QuanticNumericFacet extends LightningElement {
     this.end = this.filterState?.range?.end?.toString();
   }
 
+
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticNumericFacet/quanticNumericFacet.js:251]: extra line?

File: packages/quantic/force-app/main/default/lwc/quanticNumericFacet/quanticNumericFacet.js [hunk 3]
```diff
@@ -303,6 +309,23 @@ export default class QuanticNumericFacet extends LightningElement {
     return '';
   }
 
+  get shouldRenderInput() {
+    if(!this.withInput) {
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticNumericFacet/quanticNumericFacet.js:313]:
```suggestion
return this.withInput && this.searchStatus?.state?.hasResults || !!this.filterState?.range;
```

**Review by nathanlb — APPROVED** [review]
Good stuff!

File: packages/quantic/cypress/integration/facets/numeric-facet/numeric-facet.cypress.ts (hunk 1 above)
Comment (aelfaleh) [reply packages/quantic/cypress/integration/facets/numeric-facet/numeric-facet.cypress.ts:208]: I'm working on it

File: packages/quantic/force-app/main/default/lwc/quanticNumericFacet/quanticNumericFacet.js [hunk 4]
```diff
@@ -303,6 +309,19 @@ export default class QuanticNumericFacet extends LightningElement {
     return '';
   }
 
+  get shouldRenderInput() {
+    return this.withInput && this.searchStatus?.state?.hasResults || !!this.filterState?.range;
```
[the remark this answers is not in the document]
Comment (aelfaleh) [reply packages/quantic/force-app/main/default/lwc/quanticNumericFacet/quanticNumericFacet.js:314]: if the component markup does not include `with-input` and it is manually set in the url, will throw an error

File: packages/quantic/cypress/integration/facets/numeric-facet/numeric-facet.cypress.ts [hunk 5]
```diff
@@ -179,17 +181,35 @@ describe('Numeric Facet Test Suite', () => {
...
         Expect.clearFilterContains('Clear filter');
         Expect.numberOfSelectedCheckboxValues(1);
         Expect.numberOfIdleCheckboxValues(defaultNumberOfValues - 1);
         Expect.inputMaxEmpty();
         Expect.inputMinEmpty();
       });
+
+      scope('when selecting range with no search results', () => {
+        const min = 1;
+        const max = 10000000000;
```
[the remark this answers is not in the document]
Comment (aelfaleh) [reply packages/quantic/cypress/integration/facets/numeric-facet/numeric-facet.cypress.ts]: I used `mockSearchNoResults` for this test

**Review by lbergeron — APPROVED** [review]
